### DISCUSSION TITLE
Improved Wi-Fi/Azure IoT configuration

### DISCUSSION THREAD

**Discussion by matsujirushi** [SeeedJP/wioterminal-iotpnp-device#2] [Ideas]
https://github.com/SeeedJP/wioterminal-iotpnp-device/issues/1

> In addition to having the option to setup ssid, iotc credentials etc. over the serial console, it would be nice to use https://github.com/Seeed-Studio/Seeed_Arduino_rpcWiFiManager to have a setup experience that can be done entirely from a web browser/mobile phone.
> 
> What do you think?

**Comment by matsujirushi** [comment]
Hi benjamin,
Thanks for the suggestion.
I think that the configuration by UART is not user-friendly.

In ReButton, I was able to set up from the web page, but had the following situation.

1. There was no one to set up with mobile phone. (Because SAS KEY is long)
2. It was confusing to have so many running at the same time. (i.e. workshop/hands-on)

Here's an idea I have right now.

* Show the Wio Terminal to a PC with MSD.
* Configure Wi-Fi/AzureIoT settings in a file to MSD.
* Files are in json, yaml, and toml formats.

Reference:  
* [wio_msc_sfud](https://github.com/okini3939/wio_msc_sfud)

I'm still working on ideas.
I would appreciate your feedback.

**Reply by kartben** [reply]
I agree that in crowded environments, like workshops, this can make things complicated. 
In my opinion both UART and Wi-Fi hotspot should be supported. People familiar with the console can configure using this, but other might find it more convenient to go through the captive portal (and it does not have to be from a phone, it can be from a laptop too, where copy-paste is much easier!)

**Comment by mnakai3** [comment]
hi,
I'm trying matsujirushi and benjamin's ideas.
First, I verified the MSD, but I have confirmed that there is an existing problem that the parameters set on the device side cannot be obtained on the PC side.
https://stackoverflow.com/questions/6775951/usb-mass-storage-device-side-data-changes-not-visible-on-pc-filesystem

Therefore, I would like to be able to select the interface so that it can be configured with either MSD or CLI.
I'm thinking of achieving this by selecting the mode in two steps as shown below.

1. Reset on Button A+B+C down: enter Configuration Mode
2. selecting configuration interface by 5way-switch
    - 5Way-Left: enter MSD Mode
    - 5Way-Right: enter CLI Mode
    - 5Way-Up: enter AP Mode

**Reply by kartben** [reply]
I think it is indeed important to not enable AP mode upfront, and only do it when e.g. pressing 5-way up. For CLI, I think it should always be enabled, and there should be no need to explicitly turn it on -- for example, even if I've enabled AP mode, I might still want to quickly interact with the CLI to quickly access some settings, etc.
